controller/src/enhancements/bomb.rs

```diff
@@ -66,6 +87,27 @@ impl BombInfo {
                 .lookup(&entity_identity.entity_class_info()?)
                 .context("class name")?;
 
+            // Check if bomb is dropped
+            if class_name
+                .map(|name| name == "C_CSGameRulesProxy")
+                .unwrap_or(false)
+            {
+                /* The bomb is dropped. */
+                let rules_proxy = entity_identity
+                    .entity_ptr::<C_CSGameRulesProxy>()?
+                    .read_schema()
+                    .context("game rules missing")?;
+
+                let game_rules = rules_proxy.m_pGameRules()?.read_schema().unwrap();
+
+                if game_rules.m_bBombDropped().unwrap_or_default() {
+                    return Ok(Some(C4Info {
```

Comment:
What does ensure, that `C_CSGameRulesProxy` always comes before a potential `C_PlantedC4` entity? I'm not sure if or under what circumstances but this may lead to unexpected results.

Reply:
If the bomb is dropped it can't be planted.

Reply:
What should I do?

Reply:
You first have to loop trough all entities in order to find the `C_CSGameRulesProxy` entity.  
If found & bomb is dropped, then return C4Info with the dropped state.  
  
If not, try to find the bomb entity and get a state from that :)

Reply:
Thats what Im already doing tho? @WolverinDEV

Reply:
```
for entity_identity in entities.iter() {
            let class_name = ctx
                .class_name_cache
                .lookup(&entity_identity.entity_class_info()?)
                .context("class name")?;

            // Check if bomb is dropped
            if class_name
                .map(|name| name == "C_CSGameRulesProxy")
                .unwrap_or(false)
            {
                /* The bomb is dropped. */
                let rules_proxy = entity_identity
                    .entity_ptr::<C_CSGameRulesProxy>()?
                    .reference_schema()
                    .context("rules proxy missing")?;

                let game_rules = rules_proxy
                    .m_pGameRules()?
                    .reference_schema()
                    .context("game rules missing")?;
                ();

                if game_rules.m_bBombDropped().unwrap_or_default() {
                    return Ok(Some(C4Info {
                        bomb_site: 0,
                        state: C4State::Dropped,
                    }));
                }
            }
            
            .....
```